packages/tests/tests/local_storage_discovery.spec.ts

```diff
@@ -0,0 +1,93 @@
+import tests from "@libp2p/interface-compliance-tests/peer-discovery";
+import { LocalStoragePeerInfo } from "@waku/interfaces";
+import { LocalStorageDiscovery } from "@waku/local-storage-discovery";
+import { createLightNode, type LightNode } from "@waku/sdk";
+import { expect } from "chai";
+import { LocalStorage } from "node-localstorage";
+
+import { makeLogFileName, ServiceNode } from "../src/index.js";
+
+global.localStorage = new LocalStorage("./mock_local_storage");
+
+describe("Local Storage Discovery", function () {
+  this.timeout(25_000);
+  let waku: LightNode;
+  let serviceNode1: ServiceNode;
+  let serviceNode2: ServiceNode;
+  this.beforeEach(async function () {
+    serviceNode1 = new ServiceNode(makeLogFileName(this) + "1");
+    serviceNode2 = new ServiceNode(makeLogFileName(this) + "1");
+    await serviceNode1.start();
+    await serviceNode2.start();
+    await setPeersInLocalStorage([serviceNode1, serviceNode2]);
+  });
+
+  it("Compliance Test", async function () {
+    waku = await createLightNode();
+    await waku.start();
+    tests({
+      async setup() {
+        return new LocalStorageDiscovery(waku.libp2p.components);
+      },
+      async teardown() {}
+    });
+  });
+
+  it("Should discover peers from local storage", async function () {
```

Comment:
I think there are some improvements to do.
Discovery through local storage has two features:
- maintaining peers based on the event;
- discovering new peers from the store;

I believe only the first point makes sense to actually do `createLightNode` and even for that we do not need `nwaku` nodes as this part already `e2e` tested in other tests.

For the second point I think regular unit test is enough were, for example, we would check for:
- if parsing correct / wrong values happens as expected;
- if event is dispatched (if needed);
- if dependencies maintained: `peerStore` updated with new peer;
- etc

I can help with writhing a test if needed. 
Overall we can keep such tests in the `local-store-discovery` package.

Reply:
refactored tests to not be dependant on `sdk` or service nodes, and added a unit tests: https://github.com/waku-org/js-waku/pull/1811/commits/bf1b4ab8dddfca70137f51a262b8431d741a36ff

moved it to `local-discovery` package here: https://github.com/waku-org/js-waku/pull/1811/commits/6904f0bee5d86b9b296269802de39b37f2e02557